homeassistant/components/slide/__init__.py

```diff
@@ -133,20 +132,19 @@
     # pylint: disable=broad-except
     try:
         result = await hass.data[DOMAIN][API].login()
-    except Exception as err:
+    except (ClientConnectionError, ClientConnectorError, TimeoutError) as err:
         _LOGGER.error(
-            "Error connecting to Slide Cloud: %s, Going to retry in %s seconds",
+            "Error connecting to Slide Cloud: %s, going to retry in %s seconds",
             str(err),
             DEFAULT_RETRY,
         )
         async_call_later(hass, DEFAULT_RETRY, async_setup(hass, config))
```

Comment:
Use a lambda like here:
https://github.com/home-assistant/home-assistant/blob/ed5d3dba0e259a9648da13093a72769a2e7874c0/homeassistant/components/influxdb/__init__.py#L175

The target function for the event helper should only accept one argument.

Reply:
I tried the lambda example, but it won't work - because the async_setup is 'async' and it needs to be awaited?

Reply:
Right, then create a coroutine function that accepts the event time argument and awaits `async_setup`.

Reply:
I think I need to put it into a separate class to do this.

Thanks for you patience with me and help, as you notice, I am not a Python expert.

Reply:
Ok, found a quicker solution 👍 

```python
...
    async def retry_setup(now):
        """Retry setup if a connection/timeout happens on Slide API."""
        await async_setup(hass, config)
...
...
        async_call_later(hass, DEFAULT_RETRY, retry_setup)
        return True
...
```

Reply:
Yeah, that was what I was thinking.